services/storage/src/simcore_service_storage/dsm.py

```diff
@@ -426,3 +426,124 @@ def location_from_id(cls, location_id : str):
+            new_node_id = node_mapping.get(src_node_id)
+            if new_node_id is not None:
+                uuid_name_dict[new_node_id] = src_node['label']
+
+        # Step 1: List all objects for this project replace them with the destination object name and do a copy at the same time collect some names
+        _loop = asyncio.get_event_loop()
+        session = aiobotocore.get_session(loop=_loop)
+        async with session.create_client('s3', endpoint_url=self.s3_client.endpoint_url, aws_access_key_id=self.s3_client.access_key,
+            aws_secret_access_key=self.s3_client.secret_key) as client:
+            response = await client.list_objects_v2(Bucket=self.simcore_bucket_name, Prefix=source_folder)
+
+            if "Contents" in response:
+                for f in response['Contents']:
+                    source_object_name = f['Key']
+                    source_object_parts = Path(source_object_name).parts
+
+                    if len(source_object_parts) == 3:
+                        old_node_id = source_object_parts[1]
+                        new_node_id = node_mapping.get(old_node_id)
+                        if new_node_id is not None:
+                            old_filename = source_object_parts[2]
+                            dest_object_name = str(Path(dest_folder) / Path(new_node_id) / Path(old_filename))
+                            copy_source = {'Bucket' : self.simcore_bucket_name, 'Key': source_object_name}
+                            response = await client.copy_object(CopySource=copy_source, Bucket=self.simcore_bucket_name, Key=dest_object_name)
+
+
+            # Step 2: List all references in outputs that point to datcore and copy over
+            for node_id, node in destination_project['workbench'].items():
+                outputs = node.get("outputs")
+                if outputs is not None:
+                    for _output_key, output in outputs.items():
+                        if "store" in output and output["store"]==1:
+                            src = output["path"]
+                            dest = str(Path(dest_folder) / Path(node_id) / Path(src).name)
+                            logger.info("Need to copy %s to %s", src, dest)
+                            await self.copy_file(user_id, SIMCORE_S3_STR, dest, DATCORE_STR, src)
+                            # and change the dest project accordingly
+                            output["store"] = 0
+                            output['path'] = dest
+
+        # step 3: list files first to create fmds
+        session = aiobotocore.get_session(loop=_loop)
+        fmds = []
+        async with session.create_client('s3', endpoint_url=self.s3_client.endpoint_url, aws_access_key_id=self.s3_client.access_key,
+            aws_secret_access_key=self.s3_client.secret_key) as client:
+            response = await client.list_objects_v2(Bucket=self.simcore_bucket_name, Prefix=dest_folder+"/")
+            if 'Contents' in response:
```

Comment:
If no contents, shouldn;t you raise exception?

Reply:
I think if there are no objects, there is no content